pages/blog/graphql-deep-dive-1.mdx

```diff
@@ -0,0 +1,354 @@
+- Contribute to the [**GraphQL spec**](https://github.com/graphql/graphql-spec) with your suggestions (Some suggestions may take years to implement even if it is good, so you may need to have a lot of patience for this)
+- Contribute to the ecosystem of tools leveraging GraphQL be it with documentation, tests, features, bug fixes, feedback and what not. It will definitely help.
+- Facing a challenge with GraphQL which has not been solved before? Build your own tooling and contribute it to the community
+- Or if you want to take it to the next level and want to align your entire organization to help the GraphQL foundation, [become its member](https://foundation.graphql.org/join/) and contribute.
+
+There are a lot of small ways in which you can give back. Small or big does not matter. Every contribution counts.
+
+**Are there some case studies which can actually help me in the implementation?**
+
+Sure. While I can’t list them all here, here are some:
+
+- [Netflix and GraphQL](https://netflixtechblog.com/how-netflix-scales-its-api-with-graphql-federation-part-1-ae3557c187e2)
+- [Airbnb and GraphQL](https://medium.com/airbnb-engineering/how-airbnb-is-moving-10x-faster-at-scale-with-graphql-and-apollo-aa4ec92d69e2)
+- [Github and GraphQL](https://www.graphql.com/articles/graphql-at-github)
+- [Twitter and GraphQL](https://about.sourcegraph.com/graphql/graphql-at-twitter/)
+
+and you can find more [**here**](https://www.graphql.com/case-studies/)
+
+**Are there any publicly available GraphQL APIs which I can play around with?**
+
+Yes. While most of them would require you to authenticate, they are available for you to use. Some examples:
+
+- [Github GraphQL Explorer](https://developer.github.com/v4/explorer/)
+- [Gitlab GraphQL Explorer](https://gitlab.com/-/graphql-explorer)
+- [Yelp GraphQL Explorer](https://www.yelp.com/developers/graphiql)
+
+You can have a look at more like these [here](https://github.com/APIs-guru/graphql-apis) and play around with it.
+
+**I have a legacy architecture/stack as part of my organization. How do I incrementally migrate to GraphQL?**
+
+This is one of the places where GraphQL really shines. You need not move everything over at one piece. Here are some steps which might help.
+
+- First, build a Datagraph for your entire business without worrying about the underlying logic/implementation. But don’t worry too much since you can always evolve this over time.
+- Next, implement resolvers for every part of the schema in such a way that at phase 1, you just wrap your existing infrastructure with GraphQL. For instance, if your services use SOAP, you can add a GraphQL layer on top of it and calling that can all the SOAP service underneath and the client need not worry about it. You can use something like [GraphQL Mesh](https://graphql-mesh.com/) or [SOFA](https://sofa-api.com/) which can help in abstracting these.
+- Change the client implementation one by one to call the GraphQL gateway instead of the legacy service.
```

Comment:
not sure if you want, but that might be a good reference: https://the-guild.dev/blog/migrating-from-rest

Reply:
Sure. Will add the reference.